aten/src/ATen/core/ivalue_inl.h

```diff
@@ -724,8 +724,10 @@ inline IValue::IValue(c10::impl::GenericList v)
 
 template<class T> inline IValue::IValue(c10::List<T> v)
 : IValue(impl::toList<T>(std::move(v))) {}
-template<class T> inline IValue::IValue(at::ArrayRef<T> v)
-: IValue(c10::List<T>()) {
+template <
+    class T,
+    std::enable_if_t<!std::is_same<at::Dimname, T>::value, std::nullptr_t>>
```

Comment:
While this test "works", it is not as general as it could be. A more general test would be to test if IValue is constructible from T; if it is not, then it's wrong (much better than hardcoding Dimname!)